src/Plugin/facets/facet_source/ListFacetSourceDeriver.php

```diff
@@ -0,0 +1,73 @@
+<?php
+
+declare(strict_types = 1);
+
+namespace Drupal\oe_list_pages\Plugin\facets\facet_source;
+
+use Drupal\Core\Plugin\PluginBase;
+use Drupal\facets\FacetSource\FacetSourceDeriverBase;
+use Drupal\oe_list_pages\ListManager;
+use Symfony\Component\DependencyInjection\ContainerInterface;
+
+/**
+ * Derives a facet source plugin definition for every indexed content bundle.
+ */
+class ListFacetSourceDeriver extends FacetSourceDeriverBase {
+
+  /**
+   * The list manager.
+   *
+   * @var \Drupal\oe_list_pages\ListManager
+   */
+  private $listManager;
+
+  /**
+   * Constructs a new ListFacetDeriver.
+   *
+   * @param \Drupal\oe_list_pages\ListManager $listManager
+   *   The list manager.
+   */
+  public function __construct(ListManager $listManager) {
+    $this->listManager = $listManager;
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public static function create(ContainerInterface $container, $base_plugin_id) {
+    return new static(
+      $container->get('oe_list_pages.list_manager')
+    );
+  }
+
+  /**
+   * {@inheritdoc}
+   */
+  public function getDerivativeDefinitions($base_plugin_definition) {
```

Comment:
This method needs to return all derivatives, not just one. Not sure how tests for the presence of multiple derivatives pass with this. Look at `EntityActionDeriverBase` as an example.